p34/src/lib.rs

```diff
@@ -0,0 +1,288 @@
+use std::ops;
+
+#[derive(Debug, Hash, PartialEq)]
+pub struct BigUint4096 {
+    values: [u64; 64],
+}
+
+impl Default for BigUint4096 {
+    fn default() -> Self {
+        Self { values: [0; 64] }
+    }
+}
+
+impl ops::Add<BigUint4096> for BigUint4096 {
+    type Output = BigUint4096;
+
+    fn add(self, _rhs: BigUint4096) -> BigUint4096 {
+        &self + &_rhs
+    }
+}
+impl ops::Add<&BigUint4096> for &BigUint4096 {
+    type Output = BigUint4096;
+
+    fn add(self, _rhs: &BigUint4096) -> BigUint4096 {
+        let mut values = self.values;
+        // for i in 0..64 {
+        let mut carry = false;
+        for (i, item) in values.iter_mut().enumerate() {
+            let mut carry_from_first = false;
+            if carry {
+                (*item, carry_from_first) = item.overflowing_add(1);
+            }
+            let carry_from_second;
+            (*item, carry_from_second) = item.overflowing_add(_rhs.values[i]);
+            carry = carry_from_first || carry_from_second;
+        }
+        if carry {
+            panic!("attempt to add with overflow")
+        }
+        BigUint4096 { values }
+    }
+}
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub struct TryFromUint4096Error(pub(crate) ());
+
+impl TryFrom<u64> for BigUint4096 {
```

Comment:
In this case you can implement the `From` trait since any `u64` can be converted into `BigUint4096`.

Reply:
solved